p2p/discovery/server.go

```diff
@@ -27,6 +27,18 @@ func StartServer(port, id string) {
 	wg := sync.WaitGroup{}
 	wg.Add(1)
 	wg.Wait()
+
+}
+
+func StartServerTest(port, id string, nodeArr []*Node) (db *Database) {
+	udp := getUDP(port, HexToNodeID(id))
+	log.Debug("nodeid: %s", common.BytesToHex(udp.self.ID.Bytes()))
+	for _, node := range nodeArr {
+		udp.table.AddNode(node)
+	}
+
+	udp.StartServe()
+	return udp.db
```

Comment:
just discussion: 
generally, I suppose udp.StartServe() will be infinite service loop and never stop unless exit,
however you return udp.db, so I think this func name would be renamed to something like StartServerAndGetDB.